models/cohortdata.go

```diff
@@ -102,6 +110,33 @@ func (h CohortData) RetrieveHistogramDataBySourceIdAndCohortIdAndConceptIdsAndCo
 		Where("observation.value_as_number is not null")
 
 	query = QueryFilterByConceptIdsHelper(query, sourceId, filterConceptIds, omopDataSource, resultsDataSource.Schema, "unionAndIntersect.subject_id")
+	query, cancel := utils.AddTimeoutToQuery(query)
+	defer cancel()
+	meta_result := query.Scan(&cohortData)
+	return cohortData, meta_result.Error
+}
+
+func (h CohortData) RetrieveBarGraphDataBySourceIdAndCohortIdAndConceptIds(sourceId int, cohortDefinitionId int, histogramConceptId int64) ([]*OrdinalGroupData, error) {
+	var dataSourceModel = new(Source)
```

Comment:
```suggestion
func (h CohortData) RetrieveBarGraphDataBySourceIdAndCohortIdAndConceptId(sourceId int, cohortDefinitionId int, conceptId int64) ([]*OrdinalGroupData, error) {
	var dataSourceModel = new(Source)
```

Reply:
and you also need to filter on the given conceptId, correct? That seems to be missing, please check.